server/athenian/api/controllers/histograms_controller.py

```diff
@@ -28,8 +28,15 @@ async def calc_histogram_prs(request: AthenianWebRequest, body: dict) -> web.Res
     meta_ids = await get_metadata_account_ids(filt.account, request.sdb, request.cache)
     filters, repos = await compile_repos_and_devs_prs(filt.for_, request, filt.account, meta_ids)
     time_from, time_to = filt.resolve_time_from_and_to()
-    release_settings = \
-        await Settings.from_request(request, filt.account).list_release_matches(repos)
+    services = set(s for s, _ in filters)
+    res = await gather(
+        Settings.from_request(request, filt.account).list_release_matches(repos),
```

Comment:
We can chain `compile_repos_and_devs_prs(filt.for_, request, filt.account, meta_ids)` and `Settings.list_release_matches`.
```suggestion
    async def load_repos_and_release_settings():
        filters, repos = await compile_repos_and_devs_prs(filt.for_, request, filt.account, meta_ids)
        release_settings = await Settings.from_request(request, filt.account).list_release_matches(repos)
        return filters, repos, release_settings
    (filters, repos, release_settings), calculators = await gather(load_repos_and_release_settings(), *[get_calculator_for_user()])
```

Reply:
The problem is that we need `filters` for `get_calculator_for_user` since we need to build a calculator for each service.

Reply:
I see, NVM then.